### PR TITLE
frontends.tf.linalg: add support for l2_normalize

### LINKED ISSUE #5769: l2_normalize

### PR DESCRIPTION
**Pull request by hmahmood24**
Close #5769

### CODE REVIEW

**Comment by hmahmood24** [comment]
**Hello @RickSanchezStoic!**

Just a soft reminder to kindly review this PR as and when possible. Thanks you.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -78,3 +78,15 @@ def normalize(tensor, ord="euclidean", axis=None, name=None):
...
 )
+
+
+def l2_normalize(x, axis=None, epsilon=1e-12, name=None):
+    square_sum = ivy.sum(ivy.square(x), axis=axis, keepdims=True)
+    x_inv_norm = ivy.reciprocal(ivy.sqrt(ivy.maximum(square_sum, epsilon)))
+    return ivy.multiply(x, x_inv_norm)
+
+
+l2_normalize.supported_dtypes = (
```
Comment (RickSanchezStoic) [inline ivy/functional/frontends/tensorflow/linalg.py:89]: We now use decorators to assign `unsupported , supported dtypes and devices` . You are supposed to use `with_supported_dtypes` for this. Check this link https://lets-unify.ai/ivy/deep_dive/data_types.html#backend-data-type-bugs on how to add the version

**Comment by hmahmood24** [comment]
Hi @RickSanchezStoic,

Made the changes you asked. Please review and let me know. Thanks

**Comment by hmahmood24** [comment]
Also I have reviewed the logs for the nightly tests and the test failed for one backend but when I tried to reproduce that falsifying example using `@reproduce_failure('6.48.2', b'AXicY2JkYIAgEIBSUBoAALAACA==')` as specified in the logs, they ran successfully on my machine with the following output: `hypothesis.errors.DidNotReproduce: Expected the test to raise an error, but it completed successfully.` Just letting you know in case you want to reproduce this at your end to verify that the test runs fine.

**Comment by hmahmood24** [comment]
Hello @RickSanchezStoic,

Can you please review the changes I made 5 days ago? Thank you.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -109,3 +109,12 @@ def normalize(tensor, ord="euclidean", axis=None, name=None):
     "float32",
     "float64",
 )
+
+
```
Comment (RickSanchezStoic) [inline ivy/functional/frontends/tensorflow/linalg.py:113]: add this decorator `to_ivy_arrays_and_back`

**Review by RickSanchezStoic — CHANGES_REQUESTED** [review]

**Comment by RickSanchezStoic** [comment]
@hmahmood24 Can you try resolving the conflicts, otherwise it looks good to me

**Comment by hmahmood24** [comment]
> @hmahmood24 Can you try resolving the conflicts, otherwise it looks good to me

Resolved. 😊

**Pull request merged**